src/ucp/core/ucp_ep.c

```diff
@@ -827,7 +829,25 @@ ucs_status_ptr_t ucp_ep_close_nb(ucp_ep_h ep, unsigned mode)
                                     UCT_FLUSH_FLAG_LOCAL : UCT_FLUSH_FLAG_CANCEL,
                                     NULL, 0, NULL,
                                     ucp_ep_close_flushed_callback, "close");
-    if (!UCS_PTR_IS_PTR(request)) {
+
+    uct_cm_ep = ucp_ep_get_cm_uct_ep(ep);
+    if (uct_cm_ep != NULL) {
+        if (UCS_PTR_IS_PTR(request)) {
+        } else if (ep->flags & UCP_EP_FLAG_LOCAL_CONNECTED) {
+            /* flush state must be valid and queue is empty */
+            ucs_assert(ucp_ep_flush_state(ep) != NULL);
+            ucs_assert(ucs_queue_is_empty(&ucp_ep_ext_gen(ep)->flush_state.reqs));
+
+            close_req = ucp_request_get(ep->worker);
+            memset(close_req, 0, sizeof(*close_req));
```

Comment:
why?

Reply:
didn't want to add extra function to init close_req, it's simple enough and needed in one place only